cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -272,26 +286,29 @@ void AddCountDistinctKernels(ScalarAggregateFunction* func) {
 // Sum implementation
 
 template <typename ArrowType>
-struct SumImplDefault : public SumImpl<ArrowType, SimdLevel::NONE> {
+struct SumImplNative : public SumImpl<ArrowType, SimdLevel::NONE> {
```

Comment:
What does the naming `Native` here stand for? If it's to imply the simd level, maybe call it "Basic" or "Naive"?

Reply:
Same question, everything is native around here :-)

Reply:
I wanted a better word than "Default" and came up with "Native". "Basic" or "Naive" don't fully capture it. "Native" is bad also, I agree. Any suggestions or should I leave as "Default"?

Reply:
`Plain` maybe? If this doesn't look convincing to you then I'll vote `Default`.

Reply:
Rolling back to `Default` to keep this PR focused.